src/VisualStudio/Core/Def/Options/VisualStudioSettingsOptionPersister.cs

```diff
@@ -108,99 +100,136 @@ public bool TryFetch(OptionKey2 optionKey, string storageKey, out object? value)
             return false;
         }
 
-        public Optional<object?> TryReadOptionValue(OptionKey2 optionKey, string storageKey, Type storageType)
+        public Optional<object?> TryReadAndMonitorOptionValue(OptionKey2 primaryOptionKey, string primaryStorageKey, string storageKey, Type storageType, object? defaultValue)
         {
             Contract.ThrowIfNull(_settingManager);
+            ImmutableInterlocked.GetOrAdd(ref _storageKeysToMonitorForChanges, storageKey, _ => (primaryOptionKey, primaryStorageKey));
+            return TryReadOptionValue(_settingManager, storageKey, storageType, defaultValue);
+        }
 
-            RecordObservedValueToWatchForChanges(optionKey, storageKey, storageType);
+        internal static Optional<object?> TryReadOptionValue(ISettingsManager manager, string storageKey, Type storageType, object? defaultValue)
+        {
+            if (storageType == typeof(bool))
+                return Read<bool>();
 
-            if (storageType == typeof(bool) && _settingManager.TryGetValue(storageKey, out bool boolValue) == GetValueResult.Success)
-            {
-                return boolValue;
-            }
+            if (storageType == typeof(string))
+                return Read<string>();
 
-            if (storageType == typeof(bool?) && _settingManager.TryGetValue(storageKey, out bool? nullableBoolValue) == GetValueResult.Success)
-            {
-                return nullableBoolValue;
-            }
+            if (storageType == typeof(int))
+                return Read<int>();
 
-            if (storageType == typeof(int) && _settingManager.TryGetValue(storageKey, out int intValue) == GetValueResult.Success)
-            {
-                return intValue;
-            }
+            if (storageType.IsEnum)
+                return manager.TryGetValue(storageKey, out int value) == GetValueResult.Success ? Enum.ToObject(storageType, value) : default(Optional<object?>);
```

Comment:
💡 Seems like you could use `Read<int>()` here, and just handle the case of having a value to then call Enum.ToObject on the value.

Reply:
Not quite since `Read<int>` returns `Optional<object?>` so I'd need to extract the value from that again in order to call Enum.ToObject. Not worth it imo.